assets/training/model_evaluation/src/evaluators/evaluators.py

```diff
@@ -617,16 +617,32 @@ def evaluate(self, y_test, y_pred, **kwargs):
         """
         #  dataframe with 2 columns predictions and predictions appended to the conversation
         if len(y_pred.columns) > 1:
-            y_pred_formatted = [
-                list(item[ChatCompletionConstants.OUTPUT_FULL_CONVERSATION][0].values())[0]
-                for idx, item in y_pred.iterrows()
-            ]
+            logger.info("Found more than 1 col. Trying to fetch conversation.")
+            def check_item(row_item):
```

Comment:
(nit) Add typehints & docs.

Reply:
This "if" case is a special case we have added to handle the output from Model Prediction Component where chat completion predictions are returned in 2 formats. 1 column containing only predicted text, 2. containing full conversation.